### PR TITLE
CORS-3254: Update google.golang.org/api/cloudresourcemanager library version

### PR DESCRIPTION
**Pull request by bharath-b-rh**
This is a continuation of code changes made in https://github.com/openshift/installer/pull/7279 PR where cloudresourcemanager was updated in one of the internal packages and in this PR it is updated other packages using it.

Along with updating the library version, the APIs are also updated/replaced with the new offering from the library.

Make use of the constant `ClusterIDLabelFmt` defined in `pkg/constants/gcp/gcp.go` for deriving the label name added to GCP resources created for OpenShift cluster, instead of re-declaring at multiple places.

### CODE REVIEW

**Comment by r4f4** [comment]
/uncc @jhixson74 
/cc @bfournie @barbacbd 
To check if anything needs to be done on the capg side.

File: pkg/asset/installconfig/gcp/client.go [hunk 1]
```diff
@@ -513,3 +520,48 @@ func (c *Client) ValidateServiceAccountHasPermissions(ctx context.Context, proje
...
+// GetNamespacedTagValue returns the Tag Value metadata fetched using the tag's NamespacedName.
+func (c *Client) GetNamespacedTagValue(ctx context.Context, tagNamespacedName string) (*cloudresourcemanager.TagValue, error) {
+	service, err := c.getCloudResourceService(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create cloud resource service: %w", err)
+	}
+
+	tagValuesService := cloudresourcemanager.NewTagValuesService(service)
+
+	return tagValuesService.GetNamespaced().
```
Comment (barbacbd) [inline pkg/asset/installconfig/gcp/client.go:563]: Can we wrap the possible errors here?

File: pkg/asset/installconfig/gcp/client.go [hunk 2]
```diff
@@ -513,3 +520,48 @@ func (c *Client) ValidateServiceAccountHasPermissions(ctx context.Context, proje
...
+		Context(ctx).
+		Parent(fmt.Sprintf(projectParentPathFmt, projectID))
+
+	if err := effectiveTagsRequest.Pages(ctx, func(page *cloudresourcemanager.ListEffectiveTagsResponse) error {
+		for _, effectiveTag := range page.EffectiveTags {
+			effectiveTags.Insert(effectiveTag.NamespacedTagValue)
+		}
+		return nil
+	}); err != nil {
+		return nil, err
```
Comment (barbacbd) [inline pkg/asset/installconfig/gcp/client.go:548]: Wrap the error here.

File: pkg/destroy/gcp/policybinding.go [hunk 3]
```diff
@@ -2,20 +2,26 @@ package gcp
...
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
-	resourcemanager "google.golang.org/api/cloudresourcemanager/v1"
+	resourcemanager "google.golang.org/api/cloudresourcemanager/v3"
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
+const (
+	// projectNameFmt is the format string for project resource name.
+	projectNameFmt = "projects/%s"
```
Comment (barbacbd) [inline pkg/destroy/gcp/policybinding.go:16]: This is now defined multiple times, consider moving somewhere that both parts of the package can use it.

File: pkg/asset/installconfig/gcp/client.go (hunk 1 above)
Comment (bharath-b-rh) [reply pkg/asset/installconfig/gcp/client.go]: suggestion incorporated.

File: pkg/asset/installconfig/gcp/client.go (hunk 2 above)
Comment (bharath-b-rh) [reply pkg/asset/installconfig/gcp/client.go]: suggestion incorporated.

File: pkg/destroy/gcp/policybinding.go (hunk 3 above)
Comment (bharath-b-rh) [reply pkg/destroy/gcp/policybinding.go:16]: suggestion incorporated.

**Review by barbacbd — COMMENTED** [review]
/lgtm

**Pull request merged**